bin/sozo/src/commands/events.rs

```diff
@@ -55,63 +58,61 @@ impl EventsArgs {
+                    address: Some(world_diff.world_info.address),
+                    keys: self.events.as_ref().map(|e| {
+                        vec![e.iter().map(|event| starknet_keccak(event.as_bytes())).collect()]
+                    }),
+                };
 
-            let res =
-                provider.get_events(event_filter, self.continuation_token, self.chunk_size).await?;
-
-            for event in &res.events {
-                match world::Event::try_from(event) {
-                    Ok(ev) => {
-                        match_event(
-                            &ev,
-                            &world_diff,
-                            event.block_number,
-                            event.transaction_hash,
-                            &provider,
-                        )
-                        .await
-                        .unwrap_or_else(|e| {
-                            tracing::error!(?e, "Failed to process event: {:?}", ev);
-                        });
-                    }
-                    Err(e) => {
-                        tracing::error!(
-                            ?e,
-                            "Failed to parse remote world event which is supposed to be valid."
-                        );
+                let res = provider
+                    .get_events(event_filter, self.continuation_token.clone(), self.chunk_size)
+                    .await?;
+
+                for event in &res.events {
+                    match world::Event::try_from(event) {
+                        Ok(ev) => {
+                            match_event(
+                                &ev,
+                                &world_diff,
+                                event.block_number,
+                                event.transaction_hash,
+                                &provider,
+                            )
+                            .await
+                            .unwrap_or_else(|e| {
+                                tracing::error!(?e, "Failed to process event: {:?}", ev);
+                            });
+                        }
+                        Err(e) => {
+                            tracing::error!(
+                                ?e,
+                                "Failed to parse remote world event which is supposed to be valid."
+                            );
+                        }
                     }
                 }
-            }
 
-            if let Some(continuation_token) = res.continuation_token {
-                println!("Continuation token: {:?}", continuation_token);
-                println!("----------------------------------------------");
+                if let Some(continuation_token) = res.continuation_token {
+                    println!("Continuation token: {:?}", continuation_token);
+                    println!("----------------------------------------------");
+                }
+
+                current_start = chunk_end + 1;
```

Comment:
We can't just continue here, since the continuation token will indicate that there are more events to process. However in the new logic, since we're not stopping, those events would be lost.

The logic here could change to automatically fetch all the pages (as we do in the `world_from_events` function), and then display all the events.
Doing so, the continuation token is no more required to be exposed to the user. 👍